pkg/machine/applehv/stubber.go

```diff
@@ -0,0 +1,331 @@
+	mc.AppleHypervisor = new(vmconfigs.AppleHVConfig)
+	mc.AppleHypervisor.Vfkit = vfkit.VfkitHelper{}
+	bl := vfConfig.NewEFIBootloader(fmt.Sprintf("%s/efi-bl-%s", opts.Dirs.DataDir.GetPath(), opts.Name), true)
+	mc.AppleHypervisor.Vfkit.VirtualMachine = vfConfig.NewVirtualMachine(uint(mc.Resources.CPUs), mc.Resources.Memory, bl)
+
+	randPort, err := utils.GetRandomPort()
+	if err != nil {
+		return err
+	}
+	mc.AppleHypervisor.Vfkit.Endpoint = localhostURI + ":" + strconv.Itoa(randPort)
+
+	var virtiofsMounts []machine.VirtIoFs
+	for _, mnt := range mc.Mounts {
+		virtiofsMounts = append(virtiofsMounts, machine.MountToVirtIOFs(mnt))
+	}
+
+	// Populate the ignition file with virtiofs stuff
+	ignBuilder.WithUnit(generateSystemDFilesForVirtiofsMounts(virtiofsMounts)...)
+
+	return resizeDisk(mc, strongunits.GiB(mc.Resources.DiskSize))
+}
+
+func (a AppleHVStubber) GetHyperVisorVMs() ([]string, error) {
+	// not applicable for applehv
+	return nil, nil
+}
+
+func (a AppleHVStubber) MountType() vmconfigs.VolumeMountType {
+	return vmconfigs.VirtIOFS
+}
+
+func (a AppleHVStubber) MountVolumesToVM(_ *vmconfigs.MachineConfig, _ bool) error {
+	// virtiofs: nothing to do here
+	return nil
+}
+
+func (a AppleHVStubber) RemoveAndCleanMachines(_ *define.MachineDirs) error {
+	return nil
+}
+
+func (a AppleHVStubber) SetProviderAttrs(mc *vmconfigs.MachineConfig, cpus, memory *uint64, newDiskSize *strongunits.GiB) error {
+	if newDiskSize != nil {
+		if err := resizeDisk(mc, *newDiskSize); err != nil {
+			return err
+		}
+	}
+	// VFKit does not require saving memory, disk, or cpu
+	return nil
+}
+
+func (a AppleHVStubber) StartNetworking(mc *vmconfigs.MachineConfig, cmd *gvproxy.GvproxyCommand) error {
+	gvProxySock, err := mc.GVProxySocket()
+	if err != nil {
+		return err
+	}
+	// make sure it does not exist before gvproxy is called
+	if err := gvProxySock.Delete(); err != nil {
+		logrus.Error(err)
+	}
+	cmd.AddVfkitSocket(fmt.Sprintf("unixgram://%s", gvProxySock.GetPath()))
+	return nil
+}
+
+func (a AppleHVStubber) StartVM(mc *vmconfigs.MachineConfig) (func() error, func() error, error) {
+	var (
+		ignitionSocket *define.VMFile
+		gvproxyUp      bool
+	)
+	gvProxyWaitBackoff := 500 * time.Millisecond
+	gvProxyMaxBackoffAttempts := 6
+
+	// Add networking
+	netDevice, err := vfConfig.VirtioNetNew("5a:94:ef:e4:0c:ee")
+	if err != nil {
+		return nil, nil, err
+	}
+	// Set user networking with gvproxy
+
+	gvproxySocket, err := mc.GVProxySocket()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Wait on gvproxy to be running and aware
+	backoffWait := gvProxyWaitBackoff
+	logrus.Debug("checking that gvproxy is running")
+	for i := 0; i < gvProxyMaxBackoffAttempts; i++ {
+		err := unix.Access(gvproxySocket.GetPath(), unix.W_OK)
+		if err == nil {
+			gvproxyUp = true
+			break
+		}
+		time.Sleep(backoffWait)
+		backoffWait *= 2
+	}
```

Comment:
maybe turn this into a `isGvproxyUp` function? This `StartVM` function is pretty big on its own.